manifests/cli.pp

```diff
@@ -23,9 +23,8 @@
   }
 
   $jar = "${jenkins::libdir}/jenkins-cli.jar"
-  $extract_jar = "jar -xf ${jenkins::libdir}/jenkins.war WEB-INF/jenkins-cli.jar"
-  $move_jar = "mv WEB-INF/jenkins-cli.jar ${jar}"
-  $remove_dir = 'rm -rf WEB-INF'
+  $download_jar = 'wget https://localhost/jnlpJars/jenkins-cli.jar -O /tmp/jenkins-cli.jar'
```

Comment:
not everyone may have wget installed, that may add a new dependency
not everyone may have localhost in the HTTPS certificate, so tools may whine to not trust the cert
also, if someone like me has jenkins running under a prefix, the prefix needs to be taken into account

so maybe easier going directly to the jenkins server instead of via webserver in front of it? URL might look alike:

$cli_download_command http://127.0.0.1:${jenkins::port}${jenkins::prefix}/jnlpJars/jenkins-cli.jar